src/rgw/rgw_zone.h

```diff
@@ -1421,4 +1420,27 @@ class RGWPeriod
 };
 WRITE_CLASS_ENCODER(RGWPeriod)
 
+
+namespace rgw::sal { class ConfigStore; }
+
+// global state about the multisite configuration. initialized once during
+// startup and may be reinitialized by RGWRealmReloader, but is otherwise
+// immutable at runtime
+class RGWZoneConfig {
```

Comment:
How do we want this to interact with StoreManager::Config?  I added that mostly as a placeholder, so my gut reaction is that the two should be merged in some way.

Reply:
my main motivation for `RGWZoneConfig` was so that the runtime multisite configuration could live outside of `rgw::sal::Store`, but still be 'globally' accessible - i.e. we'd pass `RGWZoneConfig` into `process_request()` along with the `rgw::sal::Store`. this way we don't need to go through `rgw::sal::Zone/ZoneGroup` APIs to read this data which is constant at runtime - we just have to bootstrap it on startup and reload it on realm reconfiguration

`StoreManager::Config` currently just contains a `store_name` and `filter_name`, which i'm proposing that we generalize as part of `RGWZoneParams`. `StoreManager::Config` probably just becomes the thing that knows how to json-parse and binary-encode this generic store configuration in `RGWZoneParams`

on startup, we could:
* initialize the `rgw::sal::ConfigStore` and use it to load the `RGWZoneConfig`
* parse the store configuration from `RGWZoneParams` into `StoreManager::Config`
* pass `StoreManager::Config` into a store factory function like `StoreManager::get_storage()`, which recursively builds a tree of stores and returns the root

on 'realm reload', we'd re-read the `RGWZoneConfig` objects using its `rgw::sal::ConfigStore`, and call back into `StoreManager::get_storage()` to rebuild the tree of stores